### PR TITLE
s/test-tab-name/testgrid-tab-name

### PR DESCRIPTION
**Pull request by spiffxp**
Addresses https://github.com/kubernetes/test-infra/pull/13205#discussion_r301822654

### CODE REVIEW

File: experiment/migrate_testgrid_tabs.py [hunk 1]
```diff
@@ -172,7 +172,7 @@ def patch_prow_job(prowYaml, dashTab, dashboardName):
     if ("name" in dashTab
             and "testgrid-tab-name" not in annotation
             and dashTab["name"] != dashTab["test_group_name"]):
-        annotation["test-tab-name"] = dashTab["name"]
+        annotation["testgrid-tab-name"] = dashTab["name"]
```
Comment (spiffxp) [inline experiment/migrate_testgrid_tabs.py]: I did an in-place `sed` on the config files, but changed the script as well

**Comment by chases2** [comment]
There are four instances of conflicting names in config/jobs/kubernetes/cloud-provider-vsphere/cloud-provider-vsphere-config.yaml: the cloud version and the in-place version inherited a less-specific name from the specific dashboard tab, and are conflicting on `conformance-all`

File: config/jobs/kubernetes/cloud-provider-vsphere/cloud-provider-vsphere-config.yaml [hunk 2]
```diff
@@ -162,7 +162,7 @@ periodics:
         value: "5"
   annotations:
     testgrid-dashboards: conformance-all, conformance-cloud-provider-vsphere, vmware-conformance-cloud-provider
-    test-tab-name: periodic-latest
```
Comment (spiffxp) [inline config/jobs/kubernetes/cloud-provider-vsphere/cloud-provider-vsphere-config.yaml]: renamed out-of-tree provider tabs to `vsphere-latest`

File: config/jobs/kubernetes/cloud-provider-vsphere/cloud-provider-vsphere-config.yaml [hunk 3]
```diff
@@ -303,7 +303,7 @@ periodics:
         value: ci/latest.txt
   annotations:
     testgrid-dashboards: conformance-all, conformance-vsphere, vmware-conformance
-    test-tab-name: periodic-latest
```
Comment (spiffxp) [inline config/jobs/kubernetes/cloud-provider-vsphere/cloud-provider-vsphere-config.yaml]: renamed in-tree provider tabs to `in-tree-vsphere-latest`

**Comment by chases2** [comment]
/lgtm

Thanks for catching this!

**Pull request merged**